roles/command_line_tools/tasks/bootstrap.yml

```
@@ -0,0 +1,45 @@
+- name: Check for Xcode Command Line Tools
+  raw: stat /Library/Developer/CommandLineTools
+  register: xcode_cl_tools
+  changed_when: no
+  failed_when: xcode_cl_tools.rc not in [0, 1]
+  tags:
+    - macos_command_line_tools
+    - macos
+    - xcode
+
+- name: Install command line tools
+  when: xcode_cl_tools.stderr is search("No such file or directory")
+  block:
+    - name: Create hidden install file
+      raw: touch {{ _macos_cli_inprogress_file }}
+      tags:
+        - macos_command_line_tools
+        - macos
+        - xcode
+
+    - name: List updates
+      raw: softwareupdate -l | grep -o 'Command Line Tools for Xcode-.*' | tail -n 1
+      register: updates
+      changed_when: no
+      tags:
+        - macos_command_line_tools
+        - macos
+        - xcode
+
+    - name: Install Xcode Command Line Tools
+      raw: softwareupdate --install "{{ package_name }}"
+      vars:
+        package_name: "{{ updates.stdout | regex_search('Command Line Tools for Xcode-.*', multiline=True) | trim }}"
+      tags:
+        - macos_command_line_tools
+        - macos
+        - xcode
+
+  always:
+    - name: Remove hidden install file
+      raw: rm -rf {{ _macos_cli_inprogress_file }}
```

`--recursive` isn't really needed for a standalone file. Moreover, it's best to fail loudly if the target turns out to be a directory. `--verbose`, OTOH, could be useful for logs. Oh, and quoting is needed per comments above.
```suggestion
      raw: rm -fv {{ _macos_cli_inprogress_file | quote }}
```